server/debug.go

```diff
@@ -29,6 +29,23 @@ func (s *Server) registerDebugRoutes(g *echo.Group) {
 
 		return currentDebugState(c)
 	})
+
+	g.GET("/debug/log", func(c echo.Context) error {
+		var errorRecordList []*api.ErrorRecord
+
+		s.errorRecordRing.Do(func(p interface{}) {
```

Comment:
need to acquire a read lock

Reply:
Just for curiosity, why do we need a read lock?